### PR TITLE
[chore] Fix dead links

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Fixes all dead links found by markdown-check-links.

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ We release both core and contrib collectors with the same versions where the con
 1. The [Contrib](#releasing-opentelemetry-collector-contrib) collector.
 1. The [artifacts](#producing-the-artifacts)
 
+<!-- markdown-link-check-disable-next-line -->
```
Comment (bogdandrutu) [inline docs/release.md:10]: Why disabling this?

File: receiver/otlpreceiver/config.md [hunk 2]
```diff
@@ -5,30 +5,30 @@ Config defines configuration for OTLP receiver.
 
 ### Config
 
-| Name | Type | Default | Docs |
-| ---- | ---- | ------- | ---- |
-| protocols |[otlpreceiver-Protocols](#otlpreceiver-Protocols)| <no value> | Protocols is the configuration for the supported protocols, currently gRPC and HTTP (Proto and JSON).  |
+| Name | Type                                              | Default | Docs |
+| ---- |---------------------------------------------------| ------- | ---- |
```
Comment (bogdandrutu) [inline receiver/otlpreceiver/config.md:9]: why these changes in a PR that fixes dead links?

File: receiver/otlpreceiver/config.md [hunk 3]
```diff
@@ -5,30 +5,30 @@ Config defines configuration for OTLP receiver.
...
+| protocols | [otlpreceiver-Protocols](#otlpreceiver-protocols) | <no value> | Protocols is the configuration for the supported protocols, currently gRPC and HTTP (Proto and JSON).  |
 
 ### otlpreceiver-Protocols
 
-| Name | Type | Default | Docs |
-| ---- | ---- | ------- | ---- |
-| grpc |[configgrpc-GRPCServerSettings](#configgrpc-GRPCServerSettings)| <no value> | GRPCServerSettings defines common settings for a gRPC server configuration.  |
-| http |[confighttp-HTTPServerSettings](#confighttp-HTTPServerSettings)| <no value> | HTTPServerSettings defines settings for creating an HTTP server.  |
+| Name | Type                                                            | Default | Docs |
+| ---- |-----------------------------------------------------------------| ------- | ---- |
```
Comment (bogdandrutu) [inline receiver/otlpreceiver/config.md:15]: ditto

File: receiver/otlpreceiver/config.md [hunk 4]
```diff
@@ -5,30 +5,30 @@ Config defines configuration for OTLP receiver.
...
-| max_recv_msg_size_mib |uint64| <no value> | MaxRecvMsgSizeMiB sets the maximum size (in MiB) of messages accepted by the server.  |
-| max_concurrent_streams |uint32| <no value> | MaxConcurrentStreams sets the limit on the number of concurrent streams to each ServerTransport. It has effect only for streaming RPCs.  |
-| read_buffer_size |int| 524288 | ReadBufferSize for gRPC server. See grpc.ReadBufferSize (https://godoc.org/google.golang.org/grpc#ReadBufferSize).  |
-| write_buffer_size |int| <no value> | WriteBufferSize for gRPC server. See grpc.WriteBufferSize (https://godoc.org/google.golang.org/grpc#WriteBufferSize).  |
-| keepalive |[configgrpc-KeepaliveServerConfig](#configgrpc-KeepaliveServerConfig)| <no value> | Keepalive anchor for all the settings related to keepalive.  |
-| auth |[configauth-Authentication](#configauth-Authentication)| <no value> | Auth for this receiver  |
+| Name | Type                                                                  | Default | Docs |
+| ---- |-----------------------------------------------------------------------| ------- | ---- |
```
Comment (bogdandrutu) [inline receiver/otlpreceiver/config.md:22]: unrelated change.

File: docs/release.md (hunk 1 above)
Comment (TylerHelmuth) [reply docs/release.md:10]: For some reason the github link is returning a 403 error.  Rather than spend time trying to figure out why the github action is getting Forbidden errors when trying to access a github doc, I opted to ignore it.  We could add 403's in the to check_links_config's aliveStatusCodes, as 403 means its definitely a real link.

File: receiver/otlpreceiver/config.md (hunk 2 above)
Comment (TylerHelmuth) [reply receiver/otlpreceiver/config.md:9]: Goland likes to format tables when you open markdown files in the editor.

File: receiver/otlpreceiver/config.md [hunk 5]
```diff
@@ -5,30 +5,30 @@ Config defines configuration for OTLP receiver.
...
 ### otlpreceiver-Protocols
 
-| Name | Type | Default | Docs |
-| ---- | ---- | ------- | ---- |
-| grpc |[configgrpc-GRPCServerSettings](#configgrpc-GRPCServerSettings)| <no value> | GRPCServerSettings defines common settings for a gRPC server configuration.  |
-| http |[confighttp-HTTPServerSettings](#confighttp-HTTPServerSettings)| <no value> | HTTPServerSettings defines settings for creating an HTTP server.  |
+| Name | Type                                                            | Default | Docs |
+| ---- |-----------------------------------------------------------------| ------- | ---- |
+| grpc | [configgrpc-GRPCServerSettings](#configgrpc-grpcserversettings) | <no value> | GRPCServerSettings defines common settings for a gRPC server configuration.  |
+| http | [confighttp-HTTPServerSettings](#confighttp-httpserversettings) | <no value> | HTTPServerSettings defines settings for creating an HTTP server.  |
```
[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply receiver/otlpreceiver/config.md:17]: Ya capitalization does matter.  [More on the story here.](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10363)

File: receiver/otlpreceiver/config.md (hunk 2 above)
Comment (TylerHelmuth) [reply receiver/otlpreceiver/config.md:9]: Would you like me to submit a separate PR that only updates the table formats?

Comment (bogdandrutu) [reply receiver/otlpreceiver/config.md:9]: Yes please

File: receiver/otlpreceiver/config.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply receiver/otlpreceiver/config.md:17]: Thanks 🙂

File: receiver/otlpreceiver/config.md (hunk 2 above)
Comment (TylerHelmuth) [reply receiver/otlpreceiver/config.md:9]: @bogdandrutu [done](https://github.com/open-telemetry/opentelemetry-collector/pull/5743)

File: docs/security.md [hunk 6]
```diff
@@ -122,7 +122,7 @@ vectors including resource exhaustion.
 
 Component developers MUST default to encrypted connections (via the `insecure:
 false` configuration setting) and SHOULD leverage
-[receiver](https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/receiverhelper)
+[receiver](https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/receiver.go)
```
Comment (bogdandrutu) [inline docs/security.md:125]: config[grpc|http] packages I believe are relevant here, not the helpers to build the components.

File: docs/release.md (hunk 1 above)
Comment (bogdandrutu) [reply docs/release.md:10]: Would like to discuss this in a separate PR since is the only controversial change.

File: docs/security.md (hunk 6 above)
Comment (TylerHelmuth) [reply docs/security.md]: Updated to link to configgrpc and confighttp packages.

**Review by bogdandrutu — APPROVED** [review]
One last thing the disabling of that check, please remove it from this PR, and discuss the options.
